test_workspace/recordset.bosatsu

```diff
@@ -60,12 +106,26 @@ def equal_List(is_equal, l1, l2):
       []: False
       [h2, *r2]: is_equal(h1, h2) && equal_List(is_equal, r1, r2)
 
-def compare_shape3((compare1, compare2, compare3), Shape3(x1, x2, x3, x4), Shape3(y1, y2, y3, y4)):
-  compare1.equals(x1, y1) && compare2.equals(x2, y2) && compare3.equals(x3, y3) && compare3.equals(x4, y4)
+def equal_RowEntry(re1, re2):
+  match (re1, re2):
+    (REBool(RecordValue(x1)), REBool(RecordValue(x2))): cmp_Bool.equals(x1, x2)
+    (REInt(RecordValue(x1)), REInt(RecordValue(x2))): cmp_Int.equals(x1, x2)
+    (REString(RecordValue(x1)), REString(RecordValue(x2))): string_Order_fn.equals(x1, x2)
+    _: False
+
+def equal_rows(row1, row2):
+  equal_RowEntry.equal_List(row1, row2)
```

Comment:
`equal_rows = equal_List(equal_RowEntry)` would also work.... but maybe this is clearer.